src/IO/ReadBufferFromRemoteFS.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <Disks/IDiskRemote.h>
+#include <IO/ReadBufferFromFile.h>
+
+
+namespace DB
+{
+
+class ReadBufferFromRemoteFS : public ReadBufferFromFileBase
+{
+friend class ThreadPoolRemoteFSReader;
```

Comment:
Why we need friendship here?

Reply:
Because I made method `fetch` protected.